astropy/coordinates/sites.py

```diff
@@ -56,7 +56,7 @@ def __getitem__(self, site_name):
             close_names = get_close_matches(site_name, self._lowercase_names_to_locations)
             close_names = sorted(close_names, key=lambda x: len(x))
 
-            errmsg = ('Site "{0}" not in database. Use ``get_names()`` to see '
+            errmsg = ('Site "{0}" not in database. Use ``EarthLocation.get_site_names()`` to see '
```

Comment:
Hmm...the problem here is that it's at least *conceivable* that the registry gets used in a context other than `EarthLocation`.

This is a bit hacky of a suggestion, but maybe just catch the error in `EarthLocation.get_site_names` and do ``raise e.__class__(e.args[0].replace('get_names', 'EarthLocation.get_site_names'))``?